### PR TITLE
Fix PG migrations on deploy

### PR DESCRIPTION
**Pull request by oetherington**
This fixes running of Postgres migrations on deploy. We now do it inside a predeploy hook where we can execute it on a single instance by checking `$EB_IS_COMMAND_LEADER`. Unfortunately, we have to do some gross ElasticBeanstalk magic to make all the variables we need available, but it seems to work correctly (I've tested on the Postgres staging instance).

We're also now reading the Postgres connection string for migrations from the new environment variable `PG_MIGRATIONS_URL` which is configured directly in AWS as we don't have access to ForumCredentials in the hook. This seems a bit fragile though. I think we could fix this by moving the forum credential setup out of `runProduction.sh` and into a separate predeploy hook that runs before this one, but this is probably a job for a separate PR. 

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1203400931328139) by [Unito](https://www.unito.io)

### CODE REVIEW

**Comment by oetherington** [comment]
@jpaddison3 @Will-Howard I've requested review from both of you as I think you've both got more AWS experience/context than me - don't feel like you both have to take a look if you don't have time!

**Comment by jpaddison3** [comment]
> I think we could fix this by moving the forum credential setup out of runProduction.sh and into a separate predeploy hook that runs before this one, but this is probably a job for a separate PR.

I would say at the point where you're introducing a new config var for this now's the time.

File: .ebextensions/exportenv.config [hunk 1]
```diff
@@ -0,0 +1,6 @@
+commands:
```
[the remark this answers is not in the document]
Comment (oetherington) [reply .ebextensions/exportenv.config:1]: In the predeploy hook we read the AppUser and AppDeployDir settings, which aren't generally available for hooks. This file exports the needed settings into a file that can be read by the hooks. Apparently this is the recommended way to do it.

**Comment by oetherington** [comment]
> > I think we could fix this by moving the forum credential setup out of runProduction.sh and into a separate predeploy hook that runs before this one, but this is probably a job for a separate PR.
> 
> I would say at the point where you're introducing a new config var for this now's the time.

After spending some time doing some unpleasant AWS debugging for #6120 I realised that in AWS we actually define `MONGO_URL` and `PG_URL` manually as environment variables instead of reading from ForumCredentials anyway, so we don't actually need to add the extra `MONGO_MIGRATIONS_URL` (no code changes are needed here - it's probably nice to have the option to define it separately if we ever need to). My feeling is that it's probably fine to leave it as is for now since there's a non-trivial amount of risk involved in changing this, but let me know if you disagree.

**Comment by jpaddison3** [comment]
Sounds good

File: .ebextensions/exportenv.config (hunk 1 above)
[the remark this answers is not in the document]
Comment (jpaddison3) [reply .ebextensions/exportenv.config:1]: Comment to that effect?

**Review by jpaddison3 — APPROVED** [review]

**Comment by jpaddison3** [comment]
@Will-Howard do you want to review this or the other PRs?

**Pull request merged**